pkg/search/postgres/common.go

```diff
@@ -194,24 +195,55 @@ func parsePaginationQuery(pagination *v1.QueryPagination, schema *walker.Schema,
 
 	out := parsedPaginationQuery{}
 
-	for _, so := range pagination.GetSortOptions() {
-		dbField := queryFields[so.GetField()]
-		if dbField == nil {
-			return parsedPaginationQuery{}, errors.Errorf("field %s does not exist in table %s or connected tables", so.GetField(), schema.Table)
+	for idx, so := range pagination.GetSortOptions() {
+		var table, column string
+		if so.GetField() == searchPkg.DocID.String() {
+			table = schema.Table
+			column = schema.ID().ColumnName
+		} else {
+			dbField := queryFields[so.GetField()]
+			if dbField == nil {
+				return parsedPaginationQuery{}, errors.Errorf("field %s does not exist in table %s or connected tables", so.GetField(), schema.Table)
+			}
+			table, column = dbField.Schema.Table, dbField.ColumnName
+		}
+		if idx != 0 && so.GetSearchAfter() != "" {
+			return parsedPaginationQuery{}, errors.New("search after for pagination must be defined for only the first sort option")
 		}
+
 		out.OrderBys = append(out.OrderBys, orderByEntry{
 			Field: pgsearch.SelectQueryField{
-				SelectPath: qualifyColumn(dbField.Schema.Table, dbField.ColumnName),
-				FieldType:  dbField.DataType,
```

Comment:
Hmm, I think the `FieldType` is required for the deserialization to work

Reply:
Ah yeah, I guess it needs to be contained in the output because we deserialize it then potentially throw it away? The issue is that the ID doesn't have a walker.DataType which is why I did that

Reply:
I just assume ID is a string. This is a rare code path and can be removed once we deprecate bleve as the cost of search on a "DocID" is no different in postgres since the column must be top level

Reply:
Yeah we can also enforce that IDs are strings at the walker level

Reply:
done